### DISCUSSION TITLE
Outputting a MaskMap with CWatM

### DISCUSSION THREAD

**Discussion by abbylute** [iiasa/CWatM#105] [Q&A] [answered]
Hello,
I am doing a series of CWatM runs in the same location. For the initial run, I used a lon lat pair as the MaskMap and saved the catchment_totalend.nc file from that run. I've imported it into python using xarray and it looks fine:
![image](https://github.com/iiasa/CWatM/assets/30153868/cf3c8a89-1677-4ec0-9778-9002a004d88e)
I want to use this file as the MaskMap for future runs. So I changed the MaskMap value in the settings file to the location of this catchment_totalend.nc file and removed the line that outputs the catchment_totalend.nc file. I did not change anything else in the file. When I run the model again, the outputs are for the full extent of the MaskMap, but are not masked: 
![image](https://github.com/iiasa/CWatM/assets/30153868/2c45cc14-a950-4e48-b189-2b023fefe6be)
I have spent a lot of time looking through the documentation (including this guidance: https://github.com/iiasa/CWatM/discussions/17) but have not been able to figure out why the outputs are not masked. I've also tried substituting the rhine5min.tif mask that ships with the CWatM 5 min data and that provides masked outputs as expected. I've tried to use that file as a template to create other masks but have not been successful. Anyway, my understanding was that when the MaskMap is specified as a catchment_totalend.nc file it should provide masked outputs. I'd appreciate any guidance on this. Thanks in advance,
Abby

**Comment by mikhailsmilovic** [comment, marked as answer]
Guten Tag Abby,

When running with the outlet coordinates, at the top of the settings file, in [OPTIONS], include:
_savebasinmap = True_ 
This will output a TIF with the basin area that can then be used in MaskMap.

**Comment by abbylute** [comment]
Hi Mikhail, 
Thank you for your help. Using savebasinmap = True, I was able to get masked outputs as expected.
Cheers,
Abby
